tests/integration/specs/common/autostore.spec.js

```diff
@@ -0,0 +1,79 @@
+import { expect } from 'chai';
+import nock from 'nock';
+import { init, DataStore, DataStoreType, User, Query } from '__SDK__';
+import { collectionName } from '../config';
+import { cleanUpAppData } from '../utils';
+
+describe('AutoStore', function() {
+  const cachedDocs = [];
+
+  before(function () {
+    // Turn off nock
+    return nock.restore();
+  });
+
+  before(function() {
+    // Initialize the SDK
+    return init({
+      appKey: process.env.APP_KEY,
+      appSecret: process.env.APP_SECRET
+    });
+  });
+
+  before(function () {
+    // Remove any existing data in the collection
+    return cleanUpAppData(collectionName);
+  });
+
+  before(function() {
+    // Signup an anonymous user
+    return User.signup();
+  });
+
+  before(function() {
+    const collection = DataStore.collection(collectionName, DataStoreType.Sync);
+    return collection.save({ title: 'Cached Book' })
```

Comment:
If this hook is to be used for other tests - happy path for find() for example, it would be better if AutoStore is used